numpy/core/src/multiarray/arrayobject.c

```diff
@@ -1769,6 +1770,79 @@ array_free(PyObject * v)
     PyObject_Free(v);
 }
 
+static const char *
+_attr_name_as_utf8(PyObject *attr_name)
```

Comment:
This function is a mess, but tries to deal with the fact that on python 2, `getattr(a, b'test')` is the same as `getattr(a, u'test')`. I don't know the exact rules used for this coercion